### PR TITLE
feat(content): makeshift railgun accelerator attachment

### PR DESCRIPTION
**Pull request by yay855**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [ ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

There aren't very many early to midgame UPS weapon uses that aren't just lasers or the coilgun.

## Describe the solution

Added a makeshift railgun attachment for firearms that boosts the damage and range of a weapon but worsens its accuracy and drains some UPS with each shot.

### CODE REVIEW

**Review by RoyalFox2140 — COMMENTED** [review]
Made some recipe suggestions on the Discord to use an electrolysis kit to create a powerful electromagnet and more steel/copper wire. Unsure if we should also be calling for anything like metal sawing hammering or chiseling. Will leave it up more to others, Not sure how to balance this.

**Comment by chaosvolt** [comment]
Bit belatedly, one thing with this is we may wish to add that to the existing electromagnetic weapon recipes too if we do the "magnetize something with electrolysis" idea for the gunmod. Otherwise, the basic idea of adding more wire is a good step.

The two railgun recipes for reference start here, if any components in there need grabbing to add to this: https://github.com/cataclysmbnteam/Cataclysm-BN/blob/main/data/json/recipes/weapon/ranged.json#L1064

**Comment by RoyalFox2140** [comment]
> Bit belatedly, one thing with this is we may wish to add that to the existing electromagnetic weapon recipes too if we do the "magnetize something with electrolysis" idea for the gunmod. Otherwise, the basic idea of adding more wire is a good step.
> 
> The two railgun recipes for reference start here, if any components in there need grabbing to add to this: https://github.com/cataclysmbnteam/Cataclysm-BN/blob/main/data/json/recipes/weapon/ranged.json#L1064

In that case, maybe we need to update every magnetic weapon in a separate PR, with any relevant electromagnet stuff we do.

**Comment by chaosvolt** [comment]
Approving this with just the components and tools expected to be consistent with the existing weapons then adding them to all the relevant recipes later on could be fine too. Lemme compare the component lists in the meantime I suppose...

File: data/json/recipes/weapon/mods.json [hunk 1]
```diff
@@ -869,5 +869,18 @@
...
+    "type": "recipe",
+    "result": "accelerator_gunmod",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_MODS",
+    "skill_used": "electronics",
+    "skills_required": [ [ "gun", 3 ] ],
+    "difficulty": 5,
+    "time": "20 m",
+    "book_learn": [ [ "manual_electronics", 5 ], [ "advanced_electronics", 4 ], [ "textbook_electronics", 5 ] ],
+    "using": [ [ "soldering_standard", 20 ] ],
```
Comment (chaosvolt) [inline data/json/recipes/weapon/mods.json:883]: Uses a decent chunk of wire already and the critical components all seem reasonably comparable to the light rail rifle but scaled down, so only thing I'd really say is vital to add is a couple tool qualities.
```suggestion
    "using": [ [ "soldering_standard", 20 ] ],
    "qualities": [ { "id": "SAW_M_FINE", "level": 1 }, { "id": "SCREW_FINE", "level": 1 } ],
```
Adding a processor, ram, and some computer skill to represent the railgun mod needing to be a bit more complex to work with a potentially wide range of projectile velocities and sizes might be a reasonable optional addition too.

**Comment by yay855** [comment]
There we go, took your suggestions to heart and added in some electrolysis requirement, the same tools as the rebar rifle, and a steel lump and a piece of RAM and more copper wire as requirements.

**Review by RoyalFox2140 — APPROVED** [review]
Guess we'll fix the stats in post if it's too high.

**Comment by Soadreqm** [comment]
Lead isn't magnetic, so it wouldn't actually be possible to accelerate normal bullets with a magnetic field.

**Comment by yay855** [comment]
> Lead isn't magnetic, so it wouldn't actually be possible to accelerate normal bullets with a magnetic field.

It's true that lead isn't magnetic, but most bullets these days are made using copper or steel, which are. Even the bullets ingame typically have copper as an ingredient.

**Comment by RoyalFox2140** [comment]
> > Lead isn't magnetic, so it wouldn't actually be possible to accelerate normal bullets with a magnetic field.
> 
> It's true that lead isn't magnetic, but most bullets these days are made using copper or steel, which are. Even the bullets ingame typically have copper as an ingredient.

Nah most bullets are still lead IRL, birdshot isn't though. But realism isn't a concern with this one. Would we rather actually have fun here or scrap a whole PR? Merging because I see no reason why we should care about bullet composition.

**Comment by Soadreqm** [comment]
> It's true that lead isn't magnetic, but most bullets these days are made using copper or steel, which are. Even the bullets ingame typically have copper as an ingredient.

You mean full metal jacket bullets? It's just a thin sheet of harder metal with lead under it. And copper isn't magnetic either. Some armour piercing bullets have a steel core, but it doesn't look like it's exactly common.

> But realism isn't a concern with this one. Would we rather actually have fun here or scrap a whole PR?

Things like this bother me, but I guess I'll cope.

**Comment by chaosvolt** [comment]
If there was a working design using diamagnetism instead of ferromagnetism it would work on lead, but my guess is in IRL terms the power requirement to accelerate a diamagnetic projectile would probably be impractical. On the other hand, the way this item is flavored does decrease the actual power requirement: we're not having the magnets provide all of the speed in this example, instead adding more to a projectile that's already being accelerated by conventional propellant.

If we had support for things like gunmods having variable effects, then this could at least be referenced by making the damage bonus lower for most ammo but higher for any ammo flavored as having a steel core, but that may be a bit too much hassle for something fairly niche. Only other use case I can think of for what that might enable would be to allow adding rifled chokes to shotguns that increase accuracy and range when fed slugs but turn into penalties if running regular shot.

**Comment by RoyalFox2140** [comment]
> > But realism isn't a concern with this one. Would we rather actually have fun here or scrap a whole PR?
> 
> Things like this bother me, but I guess I'll cope.

We have a robust system that allows mods to obsolete items and recipes very easy. We can't make everyone happy, but a mod is free to remove an item over grounds of realism. The game is a delicate balance between fun sci fi and realism.

**Pull request merged**